source/common/upstream/upstream_impl.h

```diff
@@ -403,16 +395,13 @@ class ClusterInfoImpl : public ClusterInfo {
   const Network::Address::InstanceConstSharedPtr source_address_;
   LoadBalancerType lb_type_;
   absl::optional<envoy::api::v2::Cluster::RingHashLbConfig> lb_ring_hash_config_;
-  Ssl::ContextManager& ssl_context_manager_;
   const bool added_via_api_;
   LoadBalancerSubsetInfoImpl lb_subset_;
   const envoy::api::v2::core::Metadata metadata_;
   const envoy::api::v2::Cluster::CommonLbConfig common_lb_config_;
   const Network::ConnectionSocket::OptionsSharedPtr cluster_socket_options_;
   const bool drain_connections_on_host_removal_;
-  Secret::SecretManager& secret_manager_;
-  Init::Manager& init_manager_;
-  Server::Configuration::TransportSocketFactoryContext factory_context_;
+  Server::Configuration::TransportSocketFactoryContextImpl factory_context_;
```

Comment:
Do we need to save this factory_context_ as class member?  
Can we create it in the stack?